client.go

```diff
@@ -881,8 +881,11 @@ func generateListCacheURL(endpoint string, opts *ListOptions) (string, error) {
 	return fmt.Sprintf("%s:%s", endpoint, hashedOpts), nil
 }
 
-func isCustomTransport(transport http.RoundTripper) bool {
-	if transport != http.DefaultTransport.(*http.Transport) {
+func isCustomTransport(hc *http.Client) bool {
```

Comment:
```suggestion
func hasCustomTransport(client *http.Client) bool {
```

How about this new function name? Technically it's no longer checking hc but the http client